tests/e2e/common/common.go

```diff
@@ -695,6 +697,105 @@ func CheckMTLSStatus(details VersionDetails, opts TestOptions, shouldWarningExis
 	}
 }
 
+func SidecarInjects() func(t *testing.T) {
+	return func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), 360*time.Second)
+		t.Cleanup(cancel)
+
+		client, err := getClient()
+		require.NoError(t, err)
+
+		deploy, err := client.AppsV1().Deployments(DaprTestNamespace).Create(ctx, &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-sidecar-injected-",
+				Namespace:    DaprTestNamespace,
+			},
+			Spec: appsv1.DeploymentSpec{
+				Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"app": "sleep"}},
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{"app": "sleep"},
+						Annotations: map[string]string{
+							"dapr.io/enabled": "true",
+							"dapr.io/app-id":  "sleep",
+						},
+					},
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name:  "sleep",
+								Image: "alpine:3.11",
+								Args:  []string{"sleep", "infinity"},
+							},
+						},
+					},
+				},
+			},
+		}, metav1.CreateOptions{})
+		require.NoError(t, err)
+
+		t.Cleanup(func() {
+			cctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+			defer cancel()
+			assert.NoError(t,
+				client.AppsV1().Deployments(DaprTestNamespace).Delete(cctx, deploy.Name, metav1.DeleteOptions{}),
+			)
+		})
+
+		perr := wait.PollImmediateUntilWithContext(ctx, time.Second, func(ctx context.Context) (bool, error) {
+			deploy, err = client.AppsV1().Deployments(DaprTestNamespace).Get(ctx, deploy.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+
+			if deploy.Status.ReadyReplicas != 1 {
+				return false, nil
+			}
+
+			replicas, err := client.AppsV1().ReplicaSets(DaprTestNamespace).List(ctx, metav1.ListOptions{})
+			if err != nil {
+				return false, err
+			}
+
+			var replica string
+			for _, r := range replicas.Items {
+				for _, owner := range r.OwnerReferences {
+					if owner.Kind == "Deployment" && owner.Name == deploy.Name {
+						replica = r.Name
+						break
+					}
+				}
+				if len(replica) > 0 {
+					break
+				}
+			}
+
+			if len(replica) == 0 {
+				return false, nil
+			}
+
+			pods, err := client.CoreV1().Pods(DaprTestNamespace).List(ctx, metav1.ListOptions{})
+			if err != nil {
+				return false, err
+			}
+
+			for _, pod := range pods.Items {
+				for _, owner := range pod.OwnerReferences {
+					if owner.Kind == "ReplicaSet" && owner.Name == replica {
+						if len(pod.Spec.Containers) != 2 || pod.Spec.Containers[1].Name != "daprd" {
+							return false, errors.New("expected injected daprd container")
```

Comment:
Hi @JoshVanL, the upgrade tests are failing with this error, could you please check?